### PR TITLE
Fix style-spec scripts to work in esm package/fix help

### PR DESCRIPTION
**Pull request by acalcutt**
Attempt to fix https://github.com/maplibre/maplibre-gl-js/issues/1963

This PR updates the syntax of the style-spec bin scripts to be ESM syntax and adds a file extension to fix the following error
`TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension "" for /root/.nvm/versions/node/v16.15.0/lib/node_modules/@maplibre/maplibre-gl-style-spec/bin/gl-style-migrate. Loading extensionless files is not supported inside of "type":"module" package.json contexts. The package.json file /root/.nvm/versions/node/v16.15.0/lib/node_modules/@maplibre/maplibre-gl-style-spec/package.json caused this "type":"module" context. Try changing /root/.nvm/versions/node/v16.15.0/lib/node_modules/@maplibre/maplibre-gl-style-spec/bin/gl-style-migrate to have a file extension. Note the "bin" field of package.json can point to a file with an extension, for example {"type":"module","bin":{"gl-style-migrate":"./bin/gl-style-migrate.js"}}`

it also adds missing help information for 'gl-style-migrate' which did not exist before. 

It also removes 'gl-style-composite' for https://github.com/maplibre/maplibre-gl-js/issues/1981

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Briefly describe the changes in this PR.
 - [X] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
The type in the package.json file of the style-spec is defined as `module` which assumes js files are esm now and not cjs.
I would recommend changing the scripts to use `import` instead of `require` and not keep them in common js format.
If this avoids the need to rename them, I think it's a better option...

**Comment by acalcutt** [comment]
I mentioned changing these extensions to cjs is probably the wrong approach in the issue thread, which is why this is still a draft. This works but those scripts should be easy to convert since they are small

I was planning to attempt to change it to ESM syntax next, I just wanted to submit what I got working last night before I went to bed.

**Comment by HarelM** [comment]
Sorry, didn't mean to jump too soon. Thanks for looking into this!

**Comment by acalcutt** [comment]
I've tested this with 
npm i -g @acalcutt/maplibre-gl-style-spec@17.0.7

and [basic-v7.json](https://github.com/mapbox/mapbox-gl-styles/blob/master/styles/basic-v7.json) as a test file

**Comment by zstadler** [comment]
I've tested all scripts with a few of styles [here](https://github.com/IsraelHikingMap/VectorMap/tree/master/Styles). They all work flawlessly.

**Comment by HarelM** [comment]
Thanks! 
Can you please add a changelog item in both style-spec and the main one and bump the version of style-spec so that we can release a new version of it?

**Comment by acalcutt** [comment]
I added the changelogs and updated the version

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,7 +27,7 @@
 - Fix the exception that would be thrown on `map.setStyle` when it is passed with transformStyle option and map is initialized without an initial style. ([#1824](https://github.com/maplibre/maplibre-gl-js/pull/1824))
 - fix issue [#1582](https://github.com/maplibre/maplibre-gl-js/issues/1582) source maps are now properly generated
 - Fix the behavior of the compass button on touch devices.
-
+- Fix errors when running style-spec bin scripts and added missing help. Removed unnecessary script 'gl-style-composite'. ([#1971](https://github.com/maplibre/maplibre-gl-js/pull/1971))
```
Comment (HarelM) [inline CHANGELOG.md:30]: Can you please move it up to not be part of a released version?

Comment (acalcutt) [reply CHANGELOG.md]: done

**Review by HarelM — APPROVED** [review]

**Pull request merged**